docs/toc

```diff
@@ -1,14 +1,31 @@
-reference-manual
+expected
     and_then
-    choice
     fail
     filter
     inspect_error
     inspect
     or_else
-    pack
     recover
-    sum
     transform_error
     transform
     value_or
+optional
+    and_then
+    fail
+    filter
+    inspect_error
+    inspect
+    or_else
+    recover
+    transform_error
+    transform
+    value_or
+choice
+    index
+    and_then
+    transform
+pack
+    index
+sum
+    index
+    
```

Comment:
I like this structure (might not like it later in the life of the project, but today it's great)

Reply:
Cool, you can always adjust it later (or i will on your request).